src/widgets/checkfields/checkphone.cpp

```diff
@@ -1,21 +1,48 @@
 #include "checkqlineedit.h"
 #include "checkphone.h"
-
-CheckPhone::CheckPhone(QWidget *w, QPushButton* btn) : CheckQLineEdit(w, btn) {
-
+#include <QDebug>
+CheckPhone::CheckPhone(QWidget *w, QPushButton* btn) : CheckUntilField(w, btn) {
+    _country = "";
 }
 
+bool CheckPhone::check(QString text)
+{
+    bool isChecked = true;
 
-bool CheckPhone::check(QString text) {
     if ( text.length() == 0 ) {
-        return false;
+        isChecked = false;
+    }
+
+    if (this->getCountry() == "FRANCE") {
+        text = text.replace(" ", "");
```

Comment:
Mmmmh. Dans les conventions, on a dit pas les this… :-/
![](http://img.photobucket.com/albums/v56/bigbadlemurtamer/luke_NOOOOOO.jpg)